velox/vector/arrow/Bridge.cpp

```diff
@@ -554,11 +562,41 @@ void exportMaps(
   Selection childRows(child.size());
   exportOffsets(vec, rows, out, pool, holder, childRows);
   holder.resizeChildren(1);
-  exportBase(child, childRows, *holder.allocateChild(0), pool);
+  exportBase(child, childRows, *holder.allocateChild(0), pool, options);
   out.n_children = 1;
   out.children = holder.getChildrenArrays();
 }
 
+template <TypeKind kind>
+void flattenAndExport(
+    const BaseVector& vec,
+    const Selection& rows,
+    ArrowArray& out,
+    memory::MemoryPool* pool,
+    VeloxToArrowBridgeHolder& holder) {
+  using NativeType = typename velox::TypeTraits<kind>::NativeType;
+  SelectivityVector allRows(vec.size());
+  DecodedVector decoded(vec, allRows);
+  auto flatVector = BaseVector::create<FlatVector<NativeType>>(
+      vec.type(), decoded.size(), pool);
+
+  if (decoded.mayHaveNulls()) {
+    allRows.applyToSelected([&](vector_size_t row) {
+      if (decoded.isNullAt(row)) {
+        flatVector->setNull(row, true);
+      } else {
+        flatVector->set(row, decoded.valueAt<NativeType>(row));
+      }
+    });
+    exportBase(*flatVector, rows, out, pool, {false, false});
+  } else {
+    allRows.applyToSelected([&](vector_size_t row) {
+      flatVector->set(row, decoded.valueAt<NativeType>(row));
+    });
+    exportFlat(*flatVector, rows, out, pool, holder);
```

Comment:
Since we specialize here for each type, why not use `exportValues/exportStrings` directly?
Or I wonder if there is another way to flatten a dictionary without having to specialize for each type.

Reply:
I mean `exportFlat` is a nice wrapper around the required switch to decide if we call  `exportValues` or `exportStrings`. Now I suppose we could add a big constexpr if (instead of the switch) since the type is coming from the template so it knows which function to call when the template is instantiated.

Reply:
The main thing about the flattening is to create a type specific flat vector that holds the values as referenced by in the dictionary - and to be able to read the type specific data from the decoded vector. If there is another option (without bloat - I mean it could be written without a template but ...) I don't see it.